transactions/WFLY-17742_SupportTransactionsDuringGracefulShutdown

```diff
@@ -0,0 +1,155 @@
+  * ~~Notifications from the Transactions subsystem should explicitly report enough information to identify the transactions that are delaying the graceful shutdown of WildFly. Moreover, notifications should report whether the transaction is in-doubt or heuristic.~~ This is already implemented in the Transactions subsystem, i.e. details about transactions that need to be recovered (automatically or manually) are reported using the existing logging mechanisms of WildFly.
+
+=== Nice-to-Have Requirements
+
+* Introducing an asynchronous graceful shutdown would enable all ServerActivity implementations to receive a pre-suspend/suspend signal concurrently. In this way, the timeout will have the same duration for all ServerActivity implementations.
+* [The sequence to suspend](https://github.com/wildfly/wildfly-core/blob/main/server/src/main/java/org/jboss/as/server/suspend/SuspendController.java#L57-L99) ServerActivity implementations during WildFly’s graceful shutdown should be Last Input First Output (LIFO), i.e. the last ServerActivity implementation that was registered during startup should be the first one to get suspended.
+
+=== Non Requirements
+
+* As discussed previously, in a cloud environment, when the hosting entity crashes, its state should be recovered by the cloud environment without the intervention of the user
+
+== Implementation Plan
+
+*Aim of the proposal*. All subsystems that have the ability to start transactions should modify their suspension handlers to properly delay WildFly’s graceful shutdown as long as there are transactions to complete. Of course, users should be notified when subsystems are delaying WildFly’s suspension. This is especially true when a negative timeout is used (i.e. indefinite graceful shutdown). Moreover, internal details like 'ServerActivity' do not need to be prominent. Part of working out such a notification would be how to consistently describe a ServerActivity to a user in an understandable manner without exposing internal technical implementation details.
+
+=== Narayana
+
+Narayana needs to expose more APIs to give third-party more information about transactions. Moreover, Narayana’s Recovery Coordinator should be developed to offer more support to complete in-doubt transactions during its suspension.
+In more detail:
+* The Transaction Manager should provide a switch to suspend the creation of new transactions
+  * This new ability should be used only when no new transactions are needed. This caveat suits cases where subsystems need to create more transactions to complete their suspensions
+* The Transaction Manager should provide the ability to wait for the completion of in-flight transactions. This can be achieved:
+  * Either by providing an API to return all in-flight transactions currently in memory (i.e. delegating the control to the Application Server)
```

Comment:
Earlier on in the proposal you asked for the number of in flight transactions but here you are asking for them all to be returned.

Reply:
As the Narayana team is still discussing how we want to modify things, this part of the proposal will be addressed as soon as an agreement will be reached. Thanks for pointing this out :-)

Reply:
The number of transactions is not needed. I will change the text here. Thanks Michael